### PR TITLE
Add the ability for RenderTest to test onPropsChanged.

### LINKED ISSUE #791: [Kotlin] RenderTester should allow you to test `onPropsChanged`

### PR DESCRIPTION
**Pull request by zach-klippenstein**
Fixes #791.

### CODE REVIEW

**Comment by zach-klippenstein** [comment]
Once this lands, rebase #788.

File: kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt [hunk 1]
```diff
@@ -230,6 +230,14 @@ interface RenderTester<PropsT, StateT, OutputT : Any, RenderingT> {
     output: EmittedOutput<Any?>? = null
   ): RenderTester<PropsT, StateT, OutputT, RenderingT>
 
+  /**
+   * TODO kdoc
```
Comment (JustinDSN) [inline kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt]: Should we do this TODO?

Comment (zach-klippenstein) [reply kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt]: Yep!

**Comment by zach-klippenstein** [comment]
I'm pretty sure this is a super dumb PR – just call `onPropsChanged` directly if you want to test it.

**Pull request closed without merging**